src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -401,6 +402,19 @@ string IDurableOrchestrationContext.StartNewOrchestration(string functionName, o
         {
             this.ThrowIfInvalidAccess();
 
+            (bool isValidTimespan, string errorMessage) = this.durabilityProvider.CheckTimeInterval(retryOptions.MaxRetryInterval);
+
+            if (!isValidTimespan)
+            {
+                throw new ArgumentException($"Invalid max retry interval duration: {errorMessage}");
```

Comment:
Consider using a simpler pattern for your exception messages, like this:

```csharp
throw new ArgumentException(errorMessage, nameof(retryOptions.MaxRetryInterval);
```

The resulting exception will be:

```
System.ArgumentException: The Azure Storage provider supports a maximum of 6 days for time-based delays
Parameter name: MaxRetryInterval
```

This would be easier to maintain while still providing the right level of detail.